src/components/drawers/PreferencesDrawer.js

```diff
@@ -578,6 +608,12 @@ class PreferencesDrawer extends Component {
 
             setting.set('engines.list', engines)
 
+            if (sabaki.modules.gtplogger.validate()) {
+                if (sabaki.attachedEngineSyncers.some(x => x != null)) {
+                    sabaki.modules.gtplogger.updatePath()
+                }
+            }
+
```

Comment:
For the first point, I've made it so it re-opens the engines tab to force the user to fix it.

Regarding the second point, we need to check if engines are attached. In `updatePath` the log file gets created on the disk, so if we call it from Preferences with no engines attached we will end up creating empty log files.
I've rearranged the code so this gets checked in updatePath now, by letting updatePath take a true/false state, so we know whether we are calling it from attaching Engines in `App` or from `PreferencesDrawer`.  This is necessary for the correct behaviour to make sure we always `validate` in either case.

Reply:
No, I don't want to give `updatePath` a flag whether it's being called from `App` or `PreferencesDrawer`. It doesn't make any sense design-wise. What I meant by the second part is: Instead of

~~~js
if (sabaki.modules.gtplogger.validate()) {
    if (sabaki.attachedEngineSyncers.some(x => x != null)) {
        sabaki.modules.gtplogger.updatePath()
    }
}
~~~

we can do

~~~js
if (sabaki.attachedEngineSyncers.some(x => x != null)) {
    if (sabaki.modules.gtplogger.validate()) {
        sabaki.modules.gtplogger.updatePath()
    }
}
~~~

thus failing silently if there are no engines attached. This way, `updatePath` can always call `validate` and there's no need for a flag.